ftdetect/minisnip.vim

```diff
@@ -0,0 +1,13 @@
+" Vim ftdetect file
+" Language:      minisnip
+" Created by:    Robert Hicks <gowther@zoho.com>
+" Last Change:   2017-08-20
+"
+" This simply sets all files that are in the defined
+" directory as 'minisnip'.
+"
+
+" get the snippets directory
+let snippet_dir = expand(g:minisnip_dir) . "*"
+
+execute 'autocmd BufRead,BufNewFile ' . snippet_dir . ' set filetype=minisnip'
```

Comment:
```
Error detected while processing /home/joe/.config/nvim/plugged/vim-minisnip/ftdetect/minisnip.vim:
line   11:
E121: Undefined variable: g:minisnip_dir
E116: Invalid arguments for function expand
E15: Invalid expression: expand(g:minisnip_dir) . "**/*"
line   13:
E121: Undefined variable: snippet_dir
E15: Invalid expression: 'autocmd BufRead,BufNewFile ' . snippet_dir . ' set filetype=minisnip'
Press ENTER or type command to continue
```

Sadly I'm not sure why this is happening.
If I do `:echo expand(g:minisnip_dir)` it comes through correctly as `/home/joe/.vim/minisnip` which is correct (though I do have it symlinked to `~/joe/programs/configs/nvim/snippets` but that shouldn't matter.

Reply:
I don't know why `g:minisnip_dir` isn't working, but in #5 I changes the documentation for this to:

    autocmd BufRead,BufNewFile */minisnip/_* setlocal filetype=minisnip

This way you don't need to specify the full path, it will work multiple directories, and it is still unlikely to be triggered accidentally.

In addition to this:

- It would be better to wrap this in an `augroup`; this way people can easily delete this, for example. See [this page](https://vi.stackexchange.com/q/9455/51) for a explanation.

- It probably also be better to use `setlocal`. I *think* that for `filetype=` it's always buffer-local, but I can never remember for which settings this is. This makes it clearer, and less prone for an accidental `set` in case someone adds something later on.

Reply:
All good points I think.